### PR TITLE
docs(examples): update `socket.io` example

### PR DESCRIPTION
**Pull request by EddyVinck**
Closes: #2860

- [x] Docs

The `"dev"` script was changed from `"remix dev"` to `"remix watch"` to solve problems when working on the application as described in #2860.

### CODE REVIEW

**Comment by EddyVinck** [comment]
Should I make the change to https://github.com/remix-run/remix/blob/main/contributors.yml in this PR or should that be a separate one?

**Review by sergiodxa — APPROVED** [review]

**Comment by machour** [comment]
Hi @EddyVinck and thank you for your contribution!
Could you rebase your PR against the `main` branch?

**Comment by EddyVinck** [comment]
@machour not sure I follow

Do you want me to point this PR at the `main` branch?

image

**Comment by sergiodxa** [comment]
> @machour not sure I follow
> 
> Do you want me to point this PR at the `main` branch?
> 
> image

Yes, all examples and docs PRs goes to main directly instead of dev

**Comment by EddyVinck** [comment]
Done @sergiodxa @machour

**Comment by machour** [comment]
Looking at files changes, you seem to have included an unwanted commit (touching the `.github/` folder)

Don't hesitate to close this PR and open a brand new one, from your fork's `main` branch to the `main` branch of the repo if it's easier for you 👍🏼

**Comment by MichaelDeBoey** [comment]
@EddyVinck You can also run `git reset --hard main && git cherry-pick 6909e15 && git cherry-pick 452d332 && git push -f` to rebase your changes onto `main` branch

**Comment by EddyVinck** [comment]
Good catch @machour, thanks!

And thanks @MichaelDeBoey that worked perfectly!

File: examples/socket.io/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "sideEffects": false,
   "scripts": {
     "build": "remix build",
-    "dev": "remix dev",
+    "dev": "remix watch",
```
Comment (MichaelDeBoey) [inline examples/socket.io/package.json:9]: This looks confusing.
Can we just keep it as is, like all other examples please?

```suggestion
    "dev": "remix dev",
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: examples/socket.io/package.json (hunk 1 above)
Comment (machour) [reply examples/socket.io/package.json:9]: This is motivated by @sergiodxa's comment: https://github.com/remix-run/remix/issues/2860#issuecomment-1102130992

**Pull request merged**